cypress/e2e/API/api_EngageSphere.cy.js

```diff
@@ -0,0 +1,142 @@
+/// <reference types="cypress" />
+
+//DÚVIDAS!!!
+//to.be.at.greaterThan(9999).and.to.be.lessThan(50000)
+//Retorna que o customer Littel  tem 10.000 funcionários e também que ele possuí menos de 50.000
+
+
+
+describe('ENGAGE SPHERE API TESTING', () => {
+
+    it('Successfully retrieves customers (e.g., checks for the 200 status code)', () => {
+        cy.request('GET', 'http://localhost:3001/customers')
+            .then((response) => {
+                expect(response.status).to.eq(200);
+                expect(response.body.pageInfo.totalCustomers).to.be.is.not.null;
+
+            });
+    })
+
+    it('Returns the correct structure of the response of the first page of customers(i.e., customers and pageInfo properties)', () => {
+        cy.request('GET', 'http://localhost:3001/customers?page=1')
+            .then((response) => {
+                expect(response.status, 'Status code').to.eq(200);
+                expect(response.body.pageInfo.totalCustomers, 'Total customers').to.be.eq(50)
+                expect(response.body.pageInfo.totalPages, 'Total pages').to.be.eq(5)
+                expect(response.body.pageInfo.currentPage, 'Current page').to.be.eq("1")
+
+
+                response.body.customers.forEach((customer) => {
+                    expect(customer, 'Customer id').to.have.property('id').that.is.a('number');
+                    expect(customer, 'Customer name').to.have.property('name').that.is.a('string');
+                    expect(customer, 'Number of employees').to.have.property('employees').that.is.a('number');
+
+                    // Verifique se contactInfo é null ou um objeto
+                    if (customer.contactInfo !== null) {
```

Comment:
Abaixo vai uma versão mais simples para este condicional.

```suggestion
                    if (customer.contactInfo) {
```